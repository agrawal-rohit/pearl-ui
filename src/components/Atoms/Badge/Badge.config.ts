export default {
  parts: ["root", "text"],
  baseStyle: {
    root: {
      borderRadius: "full",
    },
    text: {
      color: "neutral.50",
    },
  },
  sizes: {
    s: {
      root: {
        w: 20,
        h: 20,
      },
      text: {
        variant: "btn4",
      },
    },
    m: {
      root: {
        w: 25,
        h: 25,
      },
      text: {
        variant: "btn4",
      },
    },
    l: {
      root: {
        w: 30,
        h: 30,
      },
      text: {
        variant: "btn3",
      },
    },
    xl: {
      root: {
        w: 35,
        h: 35,
      },
      text: {
        variant: "btn3",
      },
    },
  },
  variants: {
    primary: {
      root: {
        backgroundColor: "primary.500",
      },
    },
    success: {
      root: {
        backgroundColor: "success.500",
      },
    },
    warning: {
      root: {
        backgroundColor: "warning.500",
      },
    },
    info: {
      root: {
        backgroundColor: "info.500",
      },
    },
    error: {
      root: {
        backgroundColor: "danger.500",
      },
    },
  },
  defaults: {
    size: "m",
    variant: "primary",
  },
};
