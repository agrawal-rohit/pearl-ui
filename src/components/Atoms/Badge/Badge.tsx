import React from "react";
import Center from "../../Atoms/Center/Center";
import { useMolecularComponentConfig } from "../../../hooks/useMolecularComponentConfig";
import Text from "../Text/Text";
import { BoxProps } from "../Box/Box";

export type BadgeProps = BoxProps & {
  /** The size of the badge */
  size?: string;
  /** The variant of the badge */
  variant?: string;
};

/** A Badge is small components typically used to communicate a numerical value or indicate the status of an item to the user. */
const Badge: React.FC<BadgeProps> = ({ children, ...rest }) => {
  const molecularProps = useMolecularComponentConfig("Badge", rest, {
    size: rest.size,
    variant: rest.variant,
  });

  const renderValue = () => {
    if (!children) return null;

    if (Array.isArray(children))
      children = (children as Array<string | number>).join("");

    if (typeof children === "number" || typeof children === "string")
      return <Text {...molecularProps.text}>{children}</Text>;
    else return React.cloneElement(children as React.ReactElement);
  };

  return <Center {...molecularProps.root}>{renderValue()}</Center>;
};

export default Badge;
