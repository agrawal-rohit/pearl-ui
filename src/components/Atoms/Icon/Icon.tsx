import React from "react";
import {
  backgroundColor,
  BackgroundColorProps,
  color,
  ColorProps,
  layout,
  LayoutProps,
  spacing,
  SpacingProps,
} from "../../../theme/src/styleFunctions";
import { RNStyle, StyleFunctionContainer } from "../../../theme/src/types";
import * as ExpoIcons from "@expo/vector-icons";
import { useComponentConfig } from "../../../hooks/useComponentConfig";

const iconStyleFunctions = [
  color,
  backgroundColor,
  spacing,
  layout,
] as StyleFunctionContainer[];

type IconProps = ColorProps &
  BackgroundColorProps &
  SpacingProps &
  LayoutProps & {
    /** Icon family that contains the icon you want to use  */
    iconFamily:
      | "AntDesign"
      | "Entypo"
      | "EvilIcons"
      | "Feather"
      | "FontAwesome"
      | "FontAwesome5"
      | "Fontisto"
      | "Foundation"
      | "Ionicons"
      | "MaterialCommunityIcons"
      | "MaterialIcons"
      | "Octicons"
      | "SimpleLineIcons"
      | "Zocial";
    /** Name of the icon as given in it's respective icon family */
    iconName: string;
    /** The size of the icon */
    size?: string;
    /** The accessibility label of the icon */
    accessibilityLabel?: string;
    style?: RNStyle;
  };

/** he `Icon` component can used to add Expo Icons to your app and customize them using style props. */
const Icon: React.FC<IconProps> = ({
  iconFamily,
  iconName,
  size = "m",
  accessibilityLabel = null,
  ...props
}) => {
  const componentSpecificProps = useComponentConfig(
    "Icon",
    props,
    {
      size: size,
    },
    iconStyleFunctions
  );

  const IconToUse = ExpoIcons[iconFamily];

  return (
    <IconToUse
      accessible={true}
      accessibilityLabel={
        accessibilityLabel ? accessibilityLabel : `${iconName} Icon`
      }
      name={iconName}
      {...componentSpecificProps}
    ></IconToUse>
  );
};

export default Icon;
