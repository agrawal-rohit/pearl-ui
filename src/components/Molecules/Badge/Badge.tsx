import React from "react";
import Text from "../../atoms/text/text";
import { MoleculeComponentProps } from "../../../theme/src/types";
import Pressable, { PressableProps } from "../../atoms/pressable/pressable";
import { pearl } from "../../../pearl";

export type BaseBadgeProps = PressableProps;

/**
 * CustomBadge is a component that renders a badge with a value.
 * The value can be a number, a string, or a React element.
 * If the value is a number or a string, it will be rendered inside a Text component.
 * If the value is a React element, it will be cloned and rendered as is.
 * If the value is an array, it will be joined into a string and rendered inside a Text component.
 * If the value is undefined, nothing will be rendered.
 */
const CustomBadge = React.forwardRef(
  (
    {
      children,
      atoms,
      ...props
    }: MoleculeComponentProps<"Badge", BaseBadgeProps>,
    ref: any
  ) => {
    // Function to render the value of the badge
    const renderValue = () => {
      if (children === undefined) return null;

      // If children is an array, join it into a string
      if (Array.isArray(children)) children = children.join("");

      // If children is a number or a string, render it inside a Text component
      if (typeof children === "number" || typeof children === "string")
        return <Text {...atoms.text}>{children}</Text>;
      // If children is a React element, clone it and render it as is
      else return React.cloneElement(children as React.ReactElement);
    };

    // Return a Pressable component with the value rendered inside it
    return (
      <Pressable
        {...atoms.box}
        ref={ref}
        alignItems="center"
        justifyContent="center"
        alignSelf="flex-start"
      >
        {renderValue()}
      </Pressable>
    );
  }
);

/** A Badge is a small component typically used to communicate a numerical value or indicate the status of an item to the user. */
const Badge = pearl<BaseBadgeProps, "molecule">(
  CustomBadge,
  {
    componentName: "Badge",
    type: "molecule",
    animatable: true,
  },
  undefined,
  {
    partForOverridenStyleProps: "box",
    partForOverridenNativeProps: "box",
    partForOverridenAnimationProps: "box",
  }
);

export type BadgeProps = React.ComponentProps<typeof Badge>;

export default Badge;
