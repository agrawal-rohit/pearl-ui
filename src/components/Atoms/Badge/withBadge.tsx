import React from "react";
import Box from "../Box/Box";
import Badge, { BadgeProps } from "./Badge";

interface BadgeOptions {
  /** The position of the badge with respect to the provided base component */
  placement?: "topLeft" | "topRight" | "bottomLeft" | "bottomRight";
  /** Amount of extra spacing to add between the badge and the base component */
  offset?: number;
}

/**
 * A higher-order component which can be used to add a badge to a component
 * @param badgeValue The value to be displayed inside the badge
 * @param options A set of options for customizing the look of the badge
 * @returns
 */
export const withBadge =
  <P extends {}>(
    badgeValue: string | number | React.ReactElement | undefined,
    options: BadgeOptions & BadgeProps = {}
  ) =>
  (WrappedComponent: React.FC<P>) =>
  (props: P) => {
    const shouldBeHidden = !badgeValue;
    const { placement = "topRight", offset = 5, ...badgeProps } = options;

    const computePositionForBadge = () => {
      const positionValue = -1 * offset;

      if (placement === "topLeft")
        return { top: positionValue, left: positionValue };

      if (placement === "topRight")
        return { top: positionValue, right: positionValue };

      if (placement === "bottomLeft")
        return { bottom: positionValue, left: positionValue };

      if (placement === "bottomRight")
        return { bottom: positionValue, right: positionValue };
    };

    return (
      <Box alignSelf="flex-start">
        <WrappedComponent {...(props as P)} />
        {!shouldBeHidden && (
          <Badge
            {...badgeProps}
            position="absolute"
            style={computePositionForBadge()}
          >
            {badgeValue}
          </Badge>
        )}
      </Box>
    ) as React.ReactElement<P>;
  };
