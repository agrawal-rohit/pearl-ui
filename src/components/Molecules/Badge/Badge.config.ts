import { MolecularComponentConfig } from "../../../theme/src/types";
import { BadgeProps } from "./Badge";

export default {
  parts: ["root", "text"],
  baseStyle: {
    root: {
      backgroundColor: "primary.500",
    },
    text: {
      color: "neutral.50",
    },
  },
  sizes: {
    s: {
      root: {
        minW: 20,
        h: 20,
      },
      text: {
        px: "2xs",
        variant: "btn4",
      },
    },
    m: {
      root: {
        minW: 25,
        h: 25,
      },
      text: {
        px: "xs",
        variant: "btn4",
      },
    },
    l: {
      root: {
        minW: 30,
        h: 30,
      },
      text: {
        px: "xs",
        variant: "btn3",
      },
    },
    xl: {
      root: {
        minW: 35,
        h: 35,
      },
      text: {
        px: "s",
        variant: "btn3",
      },
    },
  },
  variants: {
    rounded: {
      root: {
        borderRadius: "full",
      },
    },
    square: {
      root: {
        borderRadius: "m",
      },
    },
  },
  defaults: {
    size: "m",
    variant: "rounded",
  },
} as MolecularComponentConfig;
