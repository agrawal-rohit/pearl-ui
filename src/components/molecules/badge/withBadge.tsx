import React, { useCallback, useMemo, useState } from "react";
import { LayoutChangeEvent, Platform } from "react-native";
import Box from "../../atoms/box/box";
import Badge, { BadgeProps } from "./badge";

export interface WithBadgeOptions {
  /**
   * The position of the badge with respect to the provided base component
   *
   * @default "topRight"
   */
  placement?: "topLeft" | "topRight" | "bottomLeft" | "bottomRight";
  /**
   * Amount of extra spacing to add between the badge and the base component
   *
   * @default 5
   */
  offset?: number;
  /**
   * Whether the badge is visible or not
   *
   * @default false
   */
  hidden?: boolean;
}

/**
 * A higher-order component which can be used to add a badge to any target component.
 * @param badgeValue The value to be displayed inside the badge
 * @param options A set of options for customizing the look of the badge
 * @returns
 */
const withBadge =
  <P extends {}>(
    badgeValue: string | number | React.ReactElement | undefined = undefined,
    options: WithBadgeOptions & BadgeProps = {}
  ) =>
  (WrappedComponent: React.FC<P>) =>
  (props: P) => {
    // Destructure options with default values
    const {
      placement = "topRight",
      offset = 5,
      hidden = false,
      ...badgeProps
    } = options;

    // State variables to hold the width of the base component and the badge
    const [baseComponentWidth, setBaseComponentWidth] = useState(0);
    const [badgeWidth, setBadgeWidth] = useState(0);

    // Function to compute the position of the badge based on the placement option
    const computePositionForBadge = useMemo(() => {
      const positionValue = -1 * offset;

      if (placement === "topLeft")
        return { top: positionValue, left: positionValue };

      if (placement === "topRight")
        return { top: positionValue, right: positionValue };

      if (placement === "bottomLeft")
        return { bottom: positionValue, left: positionValue };

      // Return bottom right position by default
      return { bottom: positionValue, right: positionValue };
    }, [placement, offset]);

    // Function to update the width of the badge when its layout changes
    const onBadgeLayoutChange = (event: LayoutChangeEvent) => {
      const { width } = event.nativeEvent.layout;
      setBadgeWidth(width);
    };

    // Function to update the width of the base component when its layout changes
    const onBaseComponentLayoutChange = (event: LayoutChangeEvent) => {
      const { width } = event.nativeEvent.layout;
      setBaseComponentWidth(width);
    };

    // Render the wrapped component with the badge
    return (
      <Box>
        <Box alignSelf="flex-start">
          <WrappedComponent
            {...(props as P)}
            onLayout={onBaseComponentLayoutChange}
          />

          {!hidden && (
            <Badge
              {...badgeProps}
              onLayout={onBadgeLayoutChange}
              position="absolute"
              zIndex="overlay"
              style={computePositionForBadge}
            >
              {badgeValue}
            </Badge>
          )}
        </Box>
      </Box>
    ) as React.ReactElement<P>;
  };

export default withBadge;
