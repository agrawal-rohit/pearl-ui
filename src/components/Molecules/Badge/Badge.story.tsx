import React from "react";
import { storiesOf } from "@storybook/react-native";
import Screen from "../../Atoms/Screen/Screen";
import Badge from "./Badge";
import Text from "../../Atoms/Text/Text";
import Icon, { IconProps } from "../../Atoms/Icon/Icon";
import withBadge from "./withBadge";

storiesOf("Badge", module)
  .addDecorator((getStory) => <Screen>{getStory()}</Screen>)
  .add("Sizes", () => (
    <>
      <Badge size="s" mb="s">
        234+
      </Badge>
      <Badge size="m" mb="s">
        234+
      </Badge>

      <Badge size="l" mb="s">
        234+
      </Badge>

      <Badge size="xl" mb="s">
        234+
      </Badge>
    </>
  ))
  .add("Variants", () => (
    <>
      <Badge variant="rounded" mb="s">
        2
      </Badge>
      <Badge variant="square" mb="s">
        NEW
      </Badge>
    </>
  ))
  .add("Color Schemes", () => (
    <>
      <Badge colorScheme="primary" mb="s">
        2
      </Badge>
      <Badge colorScheme="success" mb="s">
        2
      </Badge>

      <Badge colorScheme="warning" mb="s">
        2
      </Badge>

      <Badge colorScheme="info" mb="s">
        2
      </Badge>

      <Badge colorScheme="danger" mb="s">
        2
      </Badge>
    </>
  ))
  .add("Content type", () => (
    <>
      <Badge mb="s"></Badge>

      <Badge mb="s">23</Badge>

      <Badge mb="s">233+</Badge>

      <Badge>
        <Icon
          iconFamily="Entypo"
          iconName="edit"
          color="neutral.50"
          size="s"
        ></Icon>
      </Badge>
    </>
  ))
  .add("Attaching to a component", () => {
    const BadgedIcon = withBadge<IconProps>(1, {
      placement: "topRight",
      size: "s",
    })(Icon);

    const BadgedIconTwo = withBadge<IconProps>(
      <Icon iconFamily="Ionicons" iconName="pencil" size="s" color="white" />,
      {
        placement: "bottomRight",
        size: "m",
      }
    )(Icon);

    return (
      <>
        <BadgedIcon
          iconFamily="FontAwesome"
          iconName="inbox"
          size="xl"
          mb="l"
          testID="inbox"
        ></BadgedIcon>

        <BadgedIconTwo
          iconFamily="FontAwesome"
          iconName="inbox"
          size="xl"
        ></BadgedIconTwo>
      </>
    );
  });
