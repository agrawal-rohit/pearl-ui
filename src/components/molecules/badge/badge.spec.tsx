import React from "react";
import { render } from "@testing-library/react-native";
import Badge from "./badge";
import { ThemeProvider } from "../../../theme/src/theme-context";
import Icon, { IconProps } from "../../atoms/icon/icon";
import withBadge from "./withBadge";

jest.useFakeTimers();

describe("Molecules/Badge", () => {
  it("passes the snapshot test for different sizes", () => {
    const tree = render(
      <ThemeProvider>
        <Badge size="xs">2</Badge>
        <Badge size="s">2</Badge>
        <Badge size="m">2</Badge>
        <Badge size="l">2</Badge>
      </ThemeProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("passes the snapshot test for different variants", () => {
    const tree = render(
      <ThemeProvider>
        <Badge variant="rounded" mb="s">
          2
        </Badge>
        <Badge variant="square" mb="s">
          NEW
        </Badge>
      </ThemeProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("passes the snapshot test for different color schemes", () => {
    const tree = render(
      <ThemeProvider>
        <Badge colorScheme="primary" mb="s">
          2
        </Badge>
        <Badge colorScheme="success" mb="s">
          2
        </Badge>

        <Badge colorScheme="warning" mb="s">
          2
        </Badge>

        <Badge colorScheme="info" mb="s">
          2
        </Badge>

        <Badge colorScheme="danger" mb="s">
          2
        </Badge>
      </ThemeProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("passes the snapshot test for different values types", () => {
    const tree = render(
      <ThemeProvider>
        <Badge></Badge>

        <Badge>23</Badge>

        <Badge>{2}+</Badge>

        <Badge>
          <Icon
            iconFamily="Entypo"
            iconName="edit"
            color="neutral.50"
            size="s"
          ></Icon>
        </Badge>
      </ThemeProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("withBadge HOC works as expected", () => {
    const Wrapper = () => {
      const BadgedIcon = withBadge<IconProps>("28", {
        placement: "topRight",
        size: "s",
      })(Icon);

      const BadgedIconTwo = withBadge<IconProps>(
        <Icon iconFamily="Ionicons" iconName="pencil" size="s" color="white" />,
        {
          placement: "bottomRight",
          size: "m",
        }
      )(Icon);

      return (
        <>
          <BadgedIcon
            iconFamily="FontAwesome"
            iconName="inbox"
            size="m"
            mb="l"
          ></BadgedIcon>

          <BadgedIconTwo
            iconFamily="FontAwesome"
            iconName="inbox"
            size="m"
          ></BadgedIconTwo>
        </>
      );
    };

    const root = render(
      <ThemeProvider>
        <Wrapper />
      </ThemeProvider>
    );

    const tree = root.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("withBadge HOC works as expected with different placements", () => {
    const Wrapper = () => {
      const BadgedIconTopLeft = withBadge<IconProps>(28, {
        placement: "topLeft",
        size: "s",
      })(Icon);
      const BadgedIconTopRight = withBadge<IconProps>(28, {
        placement: "topRight",
        size: "s",
      })(Icon);
      const BadgedIconBottomLeft = withBadge<IconProps>(28, {
        placement: "bottomLeft",
        size: "s",
      })(Icon);
      const BadgedIconBottomRight = withBadge<IconProps>(28, {
        placement: "bottomRight",
        size: "s",
      })(Icon);

      return (
        <>
          <BadgedIconTopLeft
            iconFamily="FontAwesome"
            iconName="inbox"
            size="m"
            mb="l"
          ></BadgedIconTopLeft>
          <BadgedIconTopRight
            iconFamily="FontAwesome"
            iconName="inbox"
            size="m"
            mb="l"
          ></BadgedIconTopRight>
          <BadgedIconBottomLeft
            iconFamily="FontAwesome"
            iconName="inbox"
            size="m"
            mb="l"
          ></BadgedIconBottomLeft>
          <BadgedIconBottomRight
            iconFamily="FontAwesome"
            iconName="inbox"
            size="m"
            mb="l"
          ></BadgedIconBottomRight>
        </>
      );
    };

    const root = render(
      <ThemeProvider>
        <Wrapper />
      </ThemeProvider>
    );

    const tree = root.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
