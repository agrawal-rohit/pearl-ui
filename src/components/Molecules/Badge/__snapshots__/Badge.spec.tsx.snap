// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecules/Badge passes the snapshot test for different color schemes 1`] = `
Array [
  <View
    colorScheme="primary"
    onLayout={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "backgroundColor": "#6A7BFF",
        "borderRadius": 1000,
        "display": "flex",
        "height": 25,
        "justifyContent": "center",
        "marginBottom": 12,
        "minWidth": 25,
      }
    }
  >
    <Text
      accessibilityRole="text"
      accessible={true}
      allowFontScaling={true}
      style={
        Object {
          "color": "#FFFFFF",
          "display": "flex",
          "fontFamily": "Poppins-Medium",
          "fontSize": 11.5,
          "fontWeight": "500",
          "lineHeight": 16.099999999999998,
          "paddingHorizontal": 8,
        }
      }
      variant="btn4"
    >
      2
    </Text>
  </View>,
  <View
    colorScheme="success"
    onLayout={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "backgroundColor": "#A2D608",
        "borderRadius": 1000,
        "display": "flex",
        "height": 25,
        "justifyContent": "center",
        "marginBottom": 12,
        "minWidth": 25,
      }
    }
  >
    <Text
      accessibilityRole="text"
      accessible={true}
      allowFontScaling={true}
      style={
        Object {
          "color": "#FFFFFF",
          "display": "flex",
          "fontFamily": "Poppins-Medium",
          "fontSize": 11.5,
          "fontWeight": "500",
          "lineHeight": 16.099999999999998,
          "paddingHorizontal": 8,
        }
      }
      variant="btn4"
    >
      2
    </Text>
  </View>,
  <View
    colorScheme="warning"
    onLayout={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "backgroundColor": "#FFBF35",
        "borderRadius": 1000,
        "display": "flex",
        "height": 25,
        "justifyContent": "center",
        "marginBottom": 12,
        "minWidth": 25,
      }
    }
  >
    <Text
      accessibilityRole="text"
      accessible={true}
      allowFontScaling={true}
      style={
        Object {
          "color": "#FFFFFF",
          "display": "flex",
          "fontFamily": "Poppins-Medium",
          "fontSize": 11.5,
          "fontWeight": "500",
          "lineHeight": 16.099999999999998,
          "paddingHorizontal": 8,
        }
      }
      variant="btn4"
    >
      2
    </Text>
  </View>,
  <View
    colorScheme="info"
    onLayout={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "backgroundColor": "#40D2F7",
        "borderRadius": 1000,
        "display": "flex",
        "height": 25,
        "justifyContent": "center",
        "marginBottom": 12,
        "minWidth": 25,
      }
    }
  >
    <Text
      accessibilityRole="text"
      accessible={true}
      allowFontScaling={true}
      style={
        Object {
          "color": "#FFFFFF",
          "display": "flex",
          "fontFamily": "Poppins-Medium",
          "fontSize": 11.5,
          "fontWeight": "500",
          "lineHeight": 16.099999999999998,
          "paddingHorizontal": 8,
        }
      }
      variant="btn4"
    >
      2
    </Text>
  </View>,
  <View
    colorScheme="danger"
    onLayout={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "backgroundColor": "#FF4B42",
        "borderRadius": 1000,
        "display": "flex",
        "height": 25,
        "justifyContent": "center",
        "marginBottom": 12,
        "minWidth": 25,
      }
    }
  >
    <Text
      accessibilityRole="text"
      accessible={true}
      allowFontScaling={true}
      style={
        Object {
          "color": "#FFFFFF",
          "display": "flex",
          "fontFamily": "Poppins-Medium",
          "fontSize": 11.5,
          "fontWeight": "500",
          "lineHeight": 16.099999999999998,
          "paddingHorizontal": 8,
        }
      }
      variant="btn4"
    >
      2
    </Text>
  </View>,
]
`;

exports[`Molecules/Badge passes the snapshot test for different sizes 1`] = `
Array [
  <View
    onLayout={[Function]}
    size="s"
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "backgroundColor": "#6A7BFF",
        "borderRadius": 1000,
        "display": "flex",
        "height": 20,
        "justifyContent": "center",
        "minWidth": 20,
      }
    }
  >
    <Text
      accessibilityRole="text"
      accessible={true}
      allowFontScaling={true}
      style={
        Object {
          "color": "#FFFFFF",
          "display": "flex",
          "fontFamily": "Poppins-Medium",
          "fontSize": 11.5,
          "fontWeight": "500",
          "lineHeight": 16.099999999999998,
          "paddingHorizontal": 4,
        }
      }
      variant="btn4"
    >
      2
    </Text>
  </View>,
  <View
    onLayout={[Function]}
    size="m"
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "backgroundColor": "#6A7BFF",
        "borderRadius": 1000,
        "display": "flex",
        "height": 25,
        "justifyContent": "center",
        "minWidth": 25,
      }
    }
  >
    <Text
      accessibilityRole="text"
      accessible={true}
      allowFontScaling={true}
      style={
        Object {
          "color": "#FFFFFF",
          "display": "flex",
          "fontFamily": "Poppins-Medium",
          "fontSize": 11.5,
          "fontWeight": "500",
          "lineHeight": 16.099999999999998,
          "paddingHorizontal": 8,
        }
      }
      variant="btn4"
    >
      2
    </Text>
  </View>,
  <View
    onLayout={[Function]}
    size="l"
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "backgroundColor": "#6A7BFF",
        "borderRadius": 1000,
        "display": "flex",
        "height": 30,
        "justifyContent": "center",
        "minWidth": 30,
      }
    }
  >
    <Text
      accessibilityRole="text"
      accessible={true}
      allowFontScaling={true}
      style={
        Object {
          "color": "#FFFFFF",
          "display": "flex",
          "fontFamily": "Poppins-Medium",
          "fontSize": 13.799999999999999,
          "fontWeight": "500",
          "lineHeight": 18.4,
          "paddingHorizontal": 8,
        }
      }
      variant="btn3"
    >
      2
    </Text>
  </View>,
  <View
    onLayout={[Function]}
    size="xl"
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "backgroundColor": "#6A7BFF",
        "borderRadius": 1000,
        "display": "flex",
        "height": 35,
        "justifyContent": "center",
        "minWidth": 35,
      }
    }
  >
    <Text
      accessibilityRole="text"
      accessible={true}
      allowFontScaling={true}
      style={
        Object {
          "color": "#FFFFFF",
          "display": "flex",
          "fontFamily": "Poppins-Medium",
          "fontSize": 13.799999999999999,
          "fontWeight": "500",
          "lineHeight": 18.4,
          "paddingHorizontal": 12,
        }
      }
      variant="btn3"
    >
      2
    </Text>
  </View>,
]
`;

exports[`Molecules/Badge passes the snapshot test for different values types 1`] = `
Array [
  <View
    onLayout={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "backgroundColor": "#6A7BFF",
        "borderRadius": 1000,
        "display": "flex",
        "height": 25,
        "justifyContent": "center",
        "minWidth": 25,
      }
    }
  />,
  <View
    onLayout={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "backgroundColor": "#6A7BFF",
        "borderRadius": 1000,
        "display": "flex",
        "height": 25,
        "justifyContent": "center",
        "minWidth": 25,
      }
    }
  >
    <Text
      accessibilityRole="text"
      accessible={true}
      allowFontScaling={true}
      style={
        Object {
          "color": "#FFFFFF",
          "display": "flex",
          "fontFamily": "Poppins-Medium",
          "fontSize": 11.5,
          "fontWeight": "500",
          "lineHeight": 16.099999999999998,
          "paddingHorizontal": 8,
        }
      }
      variant="btn4"
    >
      23
    </Text>
  </View>,
  <View
    onLayout={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "backgroundColor": "#6A7BFF",
        "borderRadius": 1000,
        "display": "flex",
        "height": 25,
        "justifyContent": "center",
        "minWidth": 25,
      }
    }
  >
    <Text
      accessibilityRole="text"
      accessible={true}
      allowFontScaling={true}
      style={
        Object {
          "color": "#FFFFFF",
          "display": "flex",
          "fontFamily": "Poppins-Medium",
          "fontSize": 11.5,
          "fontWeight": "500",
          "lineHeight": 16.099999999999998,
          "paddingHorizontal": 8,
        }
      }
      variant="btn4"
    >
      2+
    </Text>
  </View>,
  <View
    onLayout={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "backgroundColor": "#6A7BFF",
        "borderRadius": 1000,
        "display": "flex",
        "height": 25,
        "justifyContent": "center",
        "minWidth": 25,
      }
    }
  >
    <Text />
  </View>,
]
`;

exports[`Molecules/Badge passes the snapshot test for different variants 1`] = `
Array [
  <View
    onLayout={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "backgroundColor": "#6A7BFF",
        "borderRadius": 1000,
        "display": "flex",
        "height": 25,
        "justifyContent": "center",
        "marginBottom": 12,
        "minWidth": 25,
      }
    }
    variant="rounded"
  >
    <Text
      accessibilityRole="text"
      accessible={true}
      allowFontScaling={true}
      style={
        Object {
          "color": "#FFFFFF",
          "display": "flex",
          "fontFamily": "Poppins-Medium",
          "fontSize": 11.5,
          "fontWeight": "500",
          "lineHeight": 16.099999999999998,
          "paddingHorizontal": 8,
        }
      }
      variant="btn4"
    >
      2
    </Text>
  </View>,
  <View
    onLayout={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "backgroundColor": "#6A7BFF",
        "borderRadius": 8,
        "display": "flex",
        "height": 25,
        "justifyContent": "center",
        "marginBottom": 12,
        "minWidth": 25,
      }
    }
    variant="square"
  >
    <Text
      accessibilityRole="text"
      accessible={true}
      allowFontScaling={true}
      style={
        Object {
          "color": "#FFFFFF",
          "display": "flex",
          "fontFamily": "Poppins-Medium",
          "fontSize": 11.5,
          "fontWeight": "500",
          "lineHeight": 16.099999999999998,
          "paddingHorizontal": 8,
        }
      }
      variant="btn4"
    >
      NEW
    </Text>
  </View>,
]
`;

exports[`Molecules/Badge withBadge HOC works as expected 1`] = `
Array [
  <View
    onLayout={[Function]}
    style={
      Object {
        "alignSelf": "flex-start",
        "borderRadius": undefined,
        "display": "flex",
      }
    }
  >
    <Text />
    <View
      onLayout={[Function]}
      size="s"
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "flex-start",
          "backgroundColor": "#6A7BFF",
          "borderRadius": 1000,
          "display": "flex",
          "height": 20,
          "justifyContent": "center",
          "minWidth": 20,
          "position": "absolute",
          "right": -5,
          "top": -5,
        }
      }
    >
      <Text
        accessibilityRole="text"
        accessible={true}
        allowFontScaling={true}
        style={
          Object {
            "color": "#FFFFFF",
            "display": "flex",
            "fontFamily": "Poppins-Medium",
            "fontSize": 11.5,
            "fontWeight": "500",
            "lineHeight": 16.099999999999998,
            "paddingHorizontal": 4,
          }
        }
        variant="btn4"
      >
        28
      </Text>
    </View>
  </View>,
  <View
    onLayout={[Function]}
    style={
      Object {
        "alignSelf": "flex-start",
        "borderRadius": undefined,
        "display": "flex",
      }
    }
  >
    <Text />
    <View
      onLayout={[Function]}
      size="m"
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "flex-start",
          "backgroundColor": "#6A7BFF",
          "borderRadius": 1000,
          "bottom": -5,
          "display": "flex",
          "height": 25,
          "justifyContent": "center",
          "minWidth": 25,
          "position": "absolute",
          "right": -5,
        }
      }
    >
      <Text />
    </View>
  </View>,
]
`;
